Loading obj

Ekatereana opened the pull request.

Ekatereana committed:
basic classes for builder

Ekatereana committed:
parsing classes

Ekatereana committed:
fix problems with import

Ekatereana committed:
test for parsing

Ekatereana committed:
fix problem with fileReader

Ekatereana committed:
Delete LibertStatue.mtl

Ekatereana committed:
Delete Liberty-PortaBronzo-1.bmp

Ekatereana committed:
Delete LibertStatue.obj

Ekatereana committed:
Delete Liberty-Pavimentazione-1.bmp

Ekatereana committed:
Delete Liberty-DecorazioniMarmo-2.bmp

Ekatereana committed:
Delete Liberty-MattoniBasamento-1.bmp

Ekatereana committed:
Delete Liberty-GreenBronze-1.bmp

Ekatereana committed:
fix problem with fileReader

Ekatereana committed:
Merge branch 'loadingOBJ' of https://github.com/Brookoko/ProgrammingAssignment5 into loadingOBJ

Ekatereana committed:
fix problem with comments

Ekatereana committed:
fix problem with tests

Ekatereana committed:
fix problem with public variables

Brookoko commented:
![](https://github.trello.services/images/mini-trello-icon.png) [PR#6 (Loading obj)](https://trello.com/c/i8aI71bM/3-pr6-loading-obj)

Ekatereana committed:
Delete misc.xml

Ekatereana committed:
Delete modules.xml

Ekatereana committed:
Delete vcs.xml

Ekatereana committed:
Delete workspace.xml

Ekatereana committed:
add comments to (ParserObj)
add test to (BuilderTest)

Ekatereana committed:
Merge branch 'loadingOBJ' of https://github.com/Brookoko/ProgrammingAssignment5 into loadingOBJ

# Conflicts:
#	.idea/workspace.xml

Ekatereana committed:
add test for vertex part (test.com.*.builder.vertexTest)

Ekatereana committed:
Merge branch 'dev' into loadingOBJ

Ekatereana committed:
add test for vertex part (test.com.*.builder.vertexTest)

Ekatereana committed:
fix test for vertex part (test.com.*.builder.vertexTest)

Ekatereana committed:
fix test for vertex part (test.com.*.builder.vertexTest)

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/POJO/VertexFace.java:
```diff
@@ -0,0 +1,48 @@
+ */
+public class VertexFace {
+    private int index = -1;
+    private VertexGeometry vertexG = null;
```
Geometry and Texture vertex can be represented as Vector3.

Ekatereana replied:
Already done

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/POJO/VertexFace.java:
```diff
@@ -0,0 +1,48 @@
+    private int index = -1;
+    private VertexGeometry vertexG = null;
+    private VertexTexture vertexT = null;
+    private VertexNormal vertexN = null;
```
If you need vector with only 2 coordinates, use Vector3 with zero z value or write Vector2 class.

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/POJO/FacePlate.java:
```diff
@@ -0,0 +1,54 @@
+    }
+    public VertexNormal faceNormal = new VertexNormal(0, 0, 0);
+
+    public void calculateTriangleNormal() {
```
I think this function is redundant here. Face can be made of more than 3 vertices. Or if any FacePlate is made only of 3 vertices than rewrite this using cross product. And please use counterclockwise conversion to reduce misunderstandings.

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/POJO/FacePlate.java:
```diff
@@ -0,0 +1,54 @@
+    public Material material = null;
+    public Material map = null;
+
+    public FacePlate() {
```
Add constructor for creating face with given vertices, material and map. And than make them private or at least package-private

Ekatereana replied:
Done:)

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/POJO/FacePlate.java:
```diff
@@ -0,0 +1,54 @@
+    public FacePlate() {
+    }
+
+    public void add(VertexFace vertex) {
```
Move adding vertices to constructor

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/POJO/FacePlate.java:
```diff
@@ -0,0 +1,54 @@
+    public void add(VertexFace vertex) {
+        vertices.add(vertex);
+    }
+    public VertexNormal faceNormal = new VertexNormal(0, 0, 0);
```
Make it private or at least package-private. Normal calculation can be put off for some time.

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,287 @@
+        return returnArray;
+    }
+
+    public static int[] parseIntList(String list, int startIndex) {
```
It seems complicated to me. Maybe accept string with special delimiter, for instance, only with one whitespace. And rename it to something like stringToIntArray.

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,287 @@
+        return returnArray;
+    }
+
+    public static int[] parseListVertexNTuples(String list, int expectedValuesPerTuple) {
```
Why is vertices parser in string utils?)

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,287 @@
+        int itemEnd = 0;
+        String newItem = null;
+
+        while (count < listChars.length) {
```
Why don't you use split? And you can try to use stream to write it more compactly.

Ekatereana replied:
Done)

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,287 @@
+        return returnArray;
+    }
+
+    public static String[] parseWhitespaceList(String list) {
```
Use replaceAll("\\s+", " ") and than split(" ") instead.

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,287 @@
+        return mCount;
+    }
+
+    public static float[] parseFloatList(int numFloats, String list, int startIndex) {
```
Same as with parseIntList

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,287 @@
+        int itemEnd = 0;
+        int itemLength = 0;
+
+        while (count < listLength) {
```
Try to rewrite it using Array.stream

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObj.java:
```diff
@@ -0,0 +1,487 @@
+    private Logger log = Logger.getLogger(ParseObj.class.getName());
+
+    /**
+     * string indicators of parameter in .obj file
```
Write comment starting with uppercase letter and don't forget dot at the end.

Ekatereana replied:
Done:)

Ekatereana committed:
add test for vertex part (test.com.*.builder.vertexTest)

nikitavbv commented on .idea/workspace.xml:
```diff
@@ -0,0 +1,942 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Remove this file please.

nikitavbv commented on ProgrammingAssignment5.iml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Remove this file.

Ekatereana committed:
Delete ProgrammingAssignment5.iml

Ekatereana committed:
Delete workspace.xml

Ekatereana committed:
fix problem (BuilderInterface)

Ekatereana committed:
add class Vector2 (geometry)
rewrite some methods in ParseObjFile(PossesVertex)
fix problems with FacePlateFile

Ekatereana committed:
Delete VertexNormal.java

Ekatereana committed:
Delete VertexGeometry.java

Ekatereana committed:
Delete VertexTexture.java

Ekatereana committed:
Merge branch 'loadingOBJ' of https://github.com/Brookoko/ProgrammingAssignment5 into loadingOBJ

Ekatereana committed:
fix problems with (Vector2)

Ekatereana committed:
rewrite methods in (StringUtils)

Ekatereana committed:
rewrite methods in (StringUtils)

Ekatereana committed:
fix problems with test

Ekatereana committed:
Merge branch 'dev' into loadingOBJ

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,136 @@
+
+
+
+    public static String[] parseWhitespaceList(String list) {
```
This function is redundant. You can simply replace it with string.replaceAll("\\s+", " ").split(" ")

Ekatereana committed:
fix problems with test

Ekatereana committed:
Merge branch 'loadingOBJ' of https://github.com/Brookoko/ProgrammingAssignment5 into loadingOBJ

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,136 @@
+    }
+
+
+    public static Integer[] parseListVertexNTuples(String list, int expectedValuesPerTuple) {
```
Why is this function here? And also rename it. I can`t see any list in arguments. Also return VertexFace from it

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,136 @@
+        String[] vertexStrings = list.split(" ");
+        temp = vertexStrings[0].split("/");
+        boolean mark = true;
+        Integer[] parsed;
```
It is better to create VertexFace and do not store Integer

Ekatereana replied:
Ok.

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,136 @@
+        temp = vertexStrings[0].split("/");
+        boolean mark = true;
+        Integer[] parsed;
+        if (!temp[1].equals("")) {
```
This code is redundant. Simply skip normal if it is not present.

Ekatereana replied:
There is not about normal. It about texture. It is second coordinate and we always should check if it exist. To solve nullPointedExeption

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/POJO/VertexFace.java:
```diff
@@ -0,0 +1,51 @@
+ * Class that describes structure of vertex.
+ */
+public class VertexFace {
+    private int index = -1;
```
For what purpose is this variable here?

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/POJO/VertexFace.java:
```diff
@@ -0,0 +1,51 @@
+ */
+public class VertexFace {
+    private int index = -1;
+    private Vector3 vertexG = null;
```
Assignment is redundant.

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/Main.java:
```diff
@@ -1,11 +1,16 @@
 
 public class Main {
 
   public static void main(String[] args) {
```
Remove all code from here

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/BuilderInterface.java:
```diff
@@ -0,0 +1,30 @@
+package labs.introtoprogramming.lab5.object.Builder;
+
+public interface BuilderInterface {
```
For what purpose is this interface, if there is no functions?

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/POJO/Material.java:
```diff
@@ -0,0 +1,200 @@
+    private ReflectivityTransmission kd = new ReflectivityTransmission();
+    private ReflectivityTransmission ks = new ReflectivityTransmission();
+    private ReflectivityTransmission tf = new ReflectivityTransmission();
+    private int illumModel = 0;
```
Assignment is redundant

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/POJO/Material.java:
```diff
@@ -0,0 +1,200 @@
+
+import labs.introtoprogramming.lab5.object.Builder.BuilderInterface;
+
+public class Material {
```
This class has too many information in it. Simplify it. Also remove number of setters and getters. It is bad practice to put them. Move as much dependencies as possible to constructors

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/POJO/VertexFace.java:
```diff
@@ -0,0 +1,51 @@
+/**
+ * Class that describes structure of vertex.
+ */
+public class VertexFace {
```
Decrease number of setters and getters. Move dependencies to constructors.

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,507 @@
+        String line;
+
+        // read .obj file
+        while (true) {
```
Replace (line = bufferedReader.readLine()) != null)

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,507 @@
+
+import static java.util.logging.Level.*;
+
+public class ParseObjFile {
```
There is a lot of data. We need only vertices, normal and texture coordinates. It is too complicated to understand what going on here

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Builder/LoadDataFromParserObject.java:
```diff
@@ -0,0 +1,431 @@
+
+import static java.util.logging.Level.INFO;
+
+public class LoadDataFromParserObject implements BuilderInterface {
```
Why is this class separate from parser? If you want to load info from file than create child of SceneObject and load data to it. The idea was that info about material, face, texture maps are stored in Mesh object. So it is better to load all data in mesh.

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/geometry/Vector2.java:
```diff
@@ -0,0 +1,81 @@
+    private double x;
+    private double y;
+
+    public Vector2(double x, double y) {
```
Add test for this class. You can simply copy all test from Vector3

Ekatereana committed:
rewrite basic classes.
without polygon normal

Ekatereana committed:
Delete FacePlate.java

Ekatereana committed:
Delete Material.java

Ekatereana committed:
Delete ReflectivityTransmission.java

Ekatereana committed:
Delete VertexFace.java

Ekatereana committed:
Delete BuilderInterface.java

Ekatereana committed:
Delete LoadDataFromParserObject.java

Ekatereana committed:
Delete BuilderTest.java

Ekatereana committed:
Delete FacePlateTests.java

Ekatereana committed:
Merge branch 'loadingOBJ' of https://github.com/Brookoko/ProgrammingAssignment5 into loadingOBJ

Ekatereana committed:
add polygonNormal for three vertex (PolygonObject)

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/PolygonObject.java:
```diff
@@ -0,0 +1,53 @@
+
+    public void calculatePolygonNormal(){
+        if(vertices.size() == 3){
+            Vector3 fromOneToTwo = new Vector3(vertices.get(0).getVertexG().x - vertices.get(1).getVertexG().x,
```
Replace with vertices.get(1).subtract(vertices.get(0))

Ekatereana replied:
done

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/PolygonObject.java:
```diff
@@ -0,0 +1,53 @@
+                    vertices.get(0).getVertexG().y - vertices.get(2).getVertexG().y,
+                    vertices.get(0).getVertexG().z - vertices.get(2).getVertexG().z);
+
+            polygonNormal = new Vector3(fromOneToTwo.x*fromOneToThree.y + fromOneToThree.x*fromOneToTwo.y,
```
Use crossProduct function

Ekatereana replied:
done

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,89 @@
+        log.log(SEVERE, msg1);
+    }
+
+    public static int skipWhiteSpace(int mCount, char messageChars[], String errMsg) {
```
This function is only used by itself

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,89 @@
+public class StringUtils {
+    private static Logger log = Logger.getLogger(StringUtils.class.getName());
+
+    public static void printErrMsg(String methodName, String errorMsg, int mCount, char message[]) {
```
This function is only used in skipWhiteSpace

Ekatereana replied:
done

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/Main.java:
```diff
@@ -1,11 +1,9 @@
 package labs.introtoprogramming.lab5;
 
-import labs.introtoprogramming.lab5.exception.NotImplementedException;
+import java.awt.AWTException;
```
Remove any changes in main to avoid conflicts

Brookoko commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/PolygonObject.java:
```diff
@@ -0,0 +1,53 @@
+
+import java.util.ArrayList;
+
+public class PolygonObject {
```
It is better to rename it to Polygon and PolygonObject will extends SceneObject and store all polygongs. Also it is better to replace polygon in object with triangle.

Ekatereana committed:
Delete Vector2.java

Ekatereana committed:
add polygonObject (PolygonObject)
rewrite (Polygon) and (Parser)

Ekatereana committed:
add crossProduct to polygon

Ekatereana committed:
add subtract to polygon

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/Main.java:
```diff
@@ -1,11 +1,12 @@
-  public static void main(String[] args) {
-    throw new NotImplementedException();
-  }
+        public static void main(String[] args) {
```
This file should not be changed

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+    private final static String OBJ_VERTEX = "v";
+    private final static String OBJ_FACE = "f";
+
+    private ArrayList<Vector3> verticesNormals = new ArrayList<>();
```
Better change `ArrayList<Vector3>` to `List<Vector3>`, as list implementation may change over time

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+
+    File objFile = null;
+
+
```
Remove this blank line

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+
+
+    public ParseObjFile(String filename) throws FileNotFoundException, IOException {
+
```
Remove this plan line

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+     * @throws FileNotFoundException
+     * @throws IOException
+     */
+    private void parseObjFile(String objFilename) throws FileNotFoundException, IOException {
```
It is better to make this method accept `InputStream` and add another overloaded method which passes FileInputStream.

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+
+        log.log(INFO, "Loaded " + lineCount + " lines");
+    }
+
```
Remove this empty line

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+     *
+     * @param line
+     */
+
```
Remove this empty line

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+        String[] values = line.split(" ");
+        float[] result = new float[3];
+        for (int i = 2; i < values.length; i++) {
+            result[i - 2] = Float.parseFloat(values[i]);
```
Taking into account Vector3 uses Double, maybe change to Double here?

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+     * @param line
+     */
+    private void processVertexNormal(String line) {
+        String[] values = line.split(" ");
```
This code block repeats. Maybe create `parseLineToVector3(String line, int offset)`?

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+        line = line.substring(OBJ_FACE.length()).trim();
+        Integer[] vertexIndexAry = StringUtils.parseCodeLine(line, 3);
+        addPolygon(vertexIndexAry);
+
```
Empty lines

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+        int indexOfStructure = 0;
+        Vertex fv;
+
+        while (indexOfStructure + 2 < vertexIndices.length) {
```
A short explanation would be nice here

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+        Polygon polygon = new Polygon();
+
+        int indexOfStructure = 0;
+        Vertex fv;
```
This variable can be moved inside `while` loop

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,155 @@
+            polygonObject.addVertex(fv);
+            polygon.addVertex(fv);
+            indexOfStructure += 2;
+
```
Empty line

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/Polygon.java:
```diff
@@ -0,0 +1,46 @@
+import java.util.ArrayList;
+
+public class Polygon {
+    private ArrayList<Vertex> vertices = new ArrayList<>();
```
`List<Vertex>`

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/Polygon.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    public void addVertex(Vertex vertex) {
+
```
Empty line

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/Polygon.java:
```diff
@@ -0,0 +1,46 @@
+
+    public void calculatePolygonNormalTriangle(){
+        if(vertices.size() == 3){
+
```
Empty line

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/Polygon.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    public void calculatePolygonNormalTriangle(){
+        if(vertices.size() == 3){
```
If not three, then what? Maybe throw an `AssertionError`?

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/Polygon.java:
```diff
@@ -0,0 +1,46 @@
+            Vector3 fromOneToThree = vertices.get(0).getVertexG().subtract(vertices.get(2).getVertexG());
+            polygonNormal = fromOneToTwo.crossProduct(fromOneToThree);
+        }
+
```
Empty line

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/PolygonObject.java:
```diff
@@ -0,0 +1,27 @@
+import java.util.ArrayList;
+
+public class PolygonObject extends SceneObject {
+    private ArrayList<Vertex> vertices = new ArrayList<>();
```
List

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,57 @@
+public class StringUtils {
+    private static Logger log = Logger.getLogger(StringUtils.class.getName());
+
+
```
Empty line

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,57 @@
+            return null;
+        }
+        if (list.equals("")) {
+            return null;
```
Maybe use Optional? Returning null does not seem like a good idea to me.

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,57 @@
+        if (list == null) {
+            return null;
+        }
+        if (list.equals("")) {
```
isEmpty()

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,57 @@
+        }
+        String[] temp;
+        String[] vertexStrings = list.split(" ");
+        temp = vertexStrings[0].split("/");
```
String[] temp = ...

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+
+
```
Empty lines

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/Vertex.java:
```diff
@@ -0,0 +1,33 @@
+    private Vector3 vertexG = null;
+    private Vector3 vertexN = null;
+
+
```
Empty line

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/Vertex.java:
```diff
@@ -0,0 +1,33 @@
+    private Vector3 vertexN = null;
+
+
+    public void setVertexG(Vector3 vertexG) {
```
What is G, what is N? Short comment would be nice here.

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/Vertex.java:
```diff
@@ -0,0 +1,33 @@
+        this.vertexG = vertexG;
+    }
+
+
```
Empty line here and below

nikitavbv commented on src/test/java/labs/introtoprogramming/lab5/object/Parser/ParseTest.java:
```diff
@@ -0,0 +1,24 @@
+        try {
+
+                ParseObjFile obj = new ParseObjFile("LibertStatue.obj" );
+                obj.getPolygonObject().toString();
```
add some assertions to check if file is loaded correctly.

nikitavbv commented on src/test/java/labs/introtoprogramming/lab5/object/Parser/ParseTest.java:
```diff
@@ -0,0 +1,24 @@
+
+                ParseObjFile obj = new ParseObjFile("LibertStatue.obj" );
+                obj.getPolygonObject().toString();
+            } catch (java.io.FileNotFoundException e) {
```
just add `throws FileNotFoundException` to method signature. It is okay for tests.

Ekatereana committed:
rewrite input parameter in parseObj in (ParseObjFile).

Ekatereana committed:
add tests for StringUtils

Ekatereana committed:
fix testes

Ekatereana committed:
fix problem with resources file

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,142 @@
+    private List<Vector3> verticesGeometry = new ArrayList<>();
+    private PolygonObject polygonObject = null;
+
+
```
Remove this empty line

nikitavbv commented on src/main/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFile.java:
```diff
@@ -0,0 +1,142 @@
+
+        polygonObject = new PolygonObject();
+        parseObjFile(is);
+
```
Remove this empty line

nikitavbv commented on src/test/java/labs/introtoprogramming/lab5/object/Parser/ParseObjFileTest.java:
```diff
@@ -0,0 +1,19 @@
+
+
+    @Test
+    public void ParseTest() throws IOException {
```
add some assertions to this test to check if file is loaded correctly

nikitavbv commented on src/test/java/labs/introtoprogramming/lab5/object/Parser/StringUtilsTest.java:
```diff
@@ -0,0 +1,18 @@
+        String secondLine = "1/4/2 1/7/3 1/2/5";
+        Integer[] firstArr = StringUtils.parseCodeLine(firstLine);
+        Integer[] secondArr = StringUtils.parseCodeLine(secondLine);
+        assertEquals(firstArr[0], secondArr[0]);
```
it is better to compare with some provided value, like:
assertEquals("1", firstArr[0]);
assertEquals("1", secondArr[0]);
it makes more clear what is being tested

nikitavbv commented on src/test/java/labs/introtoprogramming/lab5/object/Parser/VertexTest.java:
```diff
@@ -0,0 +1,18 @@
+        vertex.setVertexG(new Vector3(1, 1, 1));
+        assertEquals(1, vertex.getVertexG().x, 0);
+        assertEquals(0, vertex.getVertexN().x, 0 );
+        vertex.toString();
```
Maybe add an assertion here?

nikitavbv commented on src/test/java/labs/introtoprogramming/lab5/object/Parser/VertexTest.java:
```diff
@@ -0,0 +1,18 @@
+        Vertex vertex = new Vertex();
+        vertex.setVertexN(new Vector3(0, 0, 0));
+        vertex.setVertexG(new Vector3(1, 1, 1));
+        assertEquals(1, vertex.getVertexG().x, 0);
```
It would be better to have non-zero value as delta. 1e-6 should be fine

Ekatereana committed:
add more assertion to ParseObjFileTest

Ekatereana committed:
rewrite main

Ekatereana committed:
rewrite main

Ekatereana committed:
rewrite main

Ekatereana committed:
rewrite main

Ekatereana committed:
Merge branch 'dev' into loadingOBJ

Ekatereana committed:
rewrite empty line

Ekatereana committed:
Merge branch 'loadingOBJ' of https://github.com/Brookoko/ProgrammingAssignment5 into loadingOBJ

Ekatereana committed:
rewrite empty line

Ekatereana committed:
Merge branch 'dev' into loadingOBJ
